tests/state_test_util.go

```diff
@@ -235,15 +235,34 @@ func (t *StateTest) RunNoVerifyWithPost(subtest StateSubtest, vmconfig vm.Config
 	}
 	msg, err := t.json.Tx.toMessage(post, baseFee)
 	if err != nil {
-		return nil, nil, common.Hash{}, err
+		return nil, nil, common.Hash{}, fmt.Errorf("%w: toMessage: %v", err, t.json.Tx)
+	}
+
+	// Try to recover tx with current signer
```

Comment:
This now matches the logic from `RunNoVerify`. This stanza was added upstream and since this function is specific to core-geth, it was missing. `types.LatestSigner` should always be able to recover the signer for all tx types and configs.